File: LeetCode150/C++/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int i=prices[0];
        
        int max1=-1;
        for(int j=1;j<n;j++){
            if(i>prices[j]){
                i=prices[j];
            }
            else{
                max1=max(prices[j]-i,max1);
            }
        }
        if(max1<0){
            return 0;
        }
        return max1;
    }
};

File: LeetCode150/Python/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n=len(prices)
        i=prices[0]
        m=-1
        for j in range(1,n):
            i=min(prices[j],i)
            m=max(m,prices[j]-i)
        
        if(m<0):
            return 0
        return m
